### ISSUE TITLE
Encrypting the jolokia.properties file for Jolokia java agent

### ISSUE THREAD

**Issue by jaibheem** [rhuss/jolokia#448]
Hello,

I am using jolokia javaagent to connect to my cassandra process and make some queries. 

`JVM_OPTS="$JVM_OPTS -javaagent:$CASSANDRA_HOME/lib/jolokia-jvm-1.6.2-agent.jar=config=/etc/cassandra/jolokia.properties"`

I am using Authentication and the credentials are stored in jolokia.properties file. However the file is human readable when stored on the server. is there a way to encrypt the contents (at least password) of the file, so the password is not human readable?

**Comment by jaibheem** [comment]
any inputs are advice here?

**Comment by rhuss** [comment]
Which kind of authentication ? If you refer to a `keystorePassword` for storing you client cert in a keystore, you can use this encrypted. See https://jolokia.org/reference/html/agents.html#agents-jvm

Otherwise, if you refer to a plain password, then the only way to secure the password a file permission. There is no extra encryption option here.

**Comment by jaibheem** [comment]
Yes, I was referring to the plain password. I can protect using file permissions, but we add encryption feature to the password so it's not visible in plaintext?

**Comment by rhuss** [comment]
If you have an idea and feel fancy to submit a pull request, I'm happy to help to integrate this into Jolokia.
